sway/commands/input/xkb_layout.c

```diff
@@ -0,0 +1,24 @@
+#define _XOPEN_SOURCE 700
+#include "sway/commands.h"
+#include "sway/input/input-manager.h"
+#include "log.h"
+
+struct cmd_results *input_cmd_xkb_layout(int argc, char **argv) {
+	sway_log(L_DEBUG, "xkb layout for device: %s", current_input_config->identifier);
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "xkb_layout", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+	if (!current_input_config) {
+		return cmd_results_new(CMD_FAILURE, "xkb_layout", "No input device defined.");
+	}
+	struct input_config *new_config =
+		new_input_config(current_input_config->identifier);
+
+	new_config->xkb_layout = strdup(argv[0]);
```

Comment:
If you only use the first arg, should the checkarg call expect exactly 1 instead of at least 1? Does it make sense to join_args on argv and use all of them?

Reply:
Same comment applies to a few files.

Reply:
Yeah I think you're right.

Reply:
None of these seem to accept spaces. When you specify multiple options/layouts, it is a comma separated list and adding spaces does not work. I will use checkarg to specify exactly 1 argument.